src/workerd/api/node/async-hooks.h

```diff
@@ -59,25 +79,85 @@ class AsyncLocalStorage final: public jsg::Object {
   kj::Own<Key> key;
 };
 
-class AsyncResource final: public jsg::Object {
+
+class AsyncResource final: public jsg::Object, public jsg::AsyncResource {
+  // The AsyncResource API allows applications to define their own Async contexts.
+  // For instance, if user code is using custom thenables instead of native promises,
+  // or user code wishes to attach event listeners to async contexts.
 public:
-  JSG_RESOURCE_TYPE(AsyncResource) {}
+  struct Options {
+    jsg::Optional<uint64_t> triggerAsyncId;
+
+    // Node.js also has an additional `requireManualDestroy` boolean option
+    // that we do not implement.
```

Comment:
If user code were to provide this option, would it be an error for us to ignore it?

Reply:
No, this is specific to the full async_hooks API that we are not implementing. We're not implementing the destroy method the option would have no meaning. I'd say it's safe for us to ignore